src/frontend/src/handler/create_index.rs

```diff
@@ -306,11 +307,26 @@ pub async fn handle_create_index(
         {
             let catalog_reader = session.env().catalog_reader().read_guard();
             let (index_schema_name, index_table_name) = Binder::resolve_table_name(name.clone())?;
-            catalog_reader.check_relation_name_duplicated(
+
+            let relation_duplicated = catalog_reader.check_relation_name_duplicated(
                 session.database(),
                 &index_schema_name,
                 &index_table_name,
-            )?;
+            );
+
+            if if_not_exists {
+                if relation_duplicated.is_err() {
```

Comment:
If there's a relation with another type (like table) with the same name, we'll also skip it instead of throwing an error. Is this expected?

Reply:
Yes, it is the same with PostgreSQL behavior.